python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -20,11 +21,60 @@ class DataProductMetadata:
     dpia_required: bool
     tags: list[str] = field(default_factory=list)
 
+    @staticmethod
+    def from_data_product_metadata_dict(metadata: dict, version, owner_id: str):
+        """
+        Expects a dict containing data product metatdata information as per the
+        required fields in the json schema at
+        https://github.com/ministryofjustice/modernisation-platform-environments/tree/main/terraform/environments/data-platform/data-product-metadata-json-schema
+
+        and should be pass version and owner id as in openmetadata.
+
+        Then populates a DataProductMetadata object with the given data.
+        """
+        new_metadata = DataProductMetadata(
+            name=metadata.get("name", "Not provided"),
```

Comment:
I think it would be better to either pass `None` or an actual value here instead of introducing a special value. It'll clutter up the UI and make our lives more difficult if we need to fetch the metadata out of OpenMetadata at any point.

You can change the `DataProductMetadata` class to make everything optional if you prefer, I think I was assuming that the update metadata call would always have a complete dictionary, but now I'm doubting that assumption.

Reply:
yeah that would be cleaner, i'll change

Reply:
in fact we would always expect there to be most of that data available, as we're enforcing it being supplied at collection. I'll only use `.get` for non required fields